react-bootstrap/index.d.ts

```diff
@@ -590,6 +590,7 @@ declare namespace ReactBootstrap {
         active?: boolean;
         noCaret?: boolean;
         eventKey?: any;
+        title?: any; // TODO: Add more specific type
     }
     type NavDropdownProps = NavDropdownBaseProps & React.HTMLProps<NavDropdown>;
```

Comment:
This is your problem here: by intersecting with `HTMLProps` you get `title?: ReactNode & string`, to which no value is assignable (except maybe `null`)